### PR TITLE
Fix SourceAnnotation ranges and refactor to `std::fmt::Display`

### LINKED ISSUE #26: Error from Snippet to DisplayList
I use this in yarte [here](https://github.com/botika/yarte/blob/master/yarte_parser/src/error.rs#L71-L127). 

With any error returns something similar to this:
```rust
DisplayList { body: [Raw(Annotation { annotation: Annotation { annotation_type: Error, id: None, label: [DisplayTextFragment { content: "", style: Emphasis }] }, source_aligned: false, continuation: false }), Raw(Origin { path: "templates/deep/more/card/hi.hbs", pos: Some((7, 21)), header_type: Initial }), Source { lineno: None, inline_marks: [], line: Empty }, Source { lineno: Some(7), inline_marks: [], line: Content { text: "<h1>Hi, {{ name }} {{ @lastname }}", range: (0, 35) } }, Source { lineno: None, inline_marks: [], 

// Label is empty
line: Annotation { 
    annotation: Annotation { 
       annotation_type: None, id: None, 
       label: [/* WHY IS EMPTY */] }, 

range: (21, 22), annotation_type: Error, annotation_part: MultilineStart } }, Source { lineno: None, inline_marks: [], line: Empty }] }
```
And as you can see the label has vanished

### PR DESCRIPTION
**Pull request by botika**
In `test_i26` what should be the output?
panic or anything other than the label disappearing?

I would panic or even refactor to TryFrom. What you prefer

Fix #26

### CODE REVIEW

**Comment by botika** [comment]
https://github.com/rust-lang/annotate-snippets-rs/pull/27/commits/92919fe995f0a34f266a97bcd8aaa88ad00ca40d takes half the time in benches

**Comment by zbraniecki** [comment]
This PR looks awesome! Thank you so much!

for `test_i26` I'd panic for now. We can improve fallibility later.

**Comment by botika** [comment]
Ok, should panic

**Comment by zbraniecki** [comment]
Can you mark this PR as draft until you're ready for review? I'm excited to review it :)

**Comment by botika** [comment]
It seems that the coverage is not correct.
Exactly  `tests/fixtures/no-color/multiline_annotation.yaml`. 

In the [report](https://travis-ci.com/github/botika/annotate-snippets-rs/jobs/300632445#L648) details:
```
thread 'test_fixtures' panicked at 'SourceAnnotation range `(23, 786)` is bigger than source length `766`', src/display_list/from_snippet.rs:190:9
```
So I think there are more bugs. So I will review that part of the logic.

**Comment by botika** [comment]
Should already be presentable and https://github.com/rust-lang/annotate-snippets-rs/pull/27/commits/90181db4d47d9f7b974f60c2974a58bf8b9ffa68 doubles performance

File: src/display_list/mod.rs [hunk 1]
```diff
@@ -32,7 +32,7 @@
 //!
 //! The above snippet has been built out of the following structure:
 //!
-//! ```
+//! ```rust,ignore
```
Comment (zbraniecki) [inline src/display_list/mod.rs]: A leftover or do you think this doc example doesn't make sense?

Comment (botika) [reply src/display_list/mod.rs]: Is leftover

**Comment by botika** [comment]
The changes are already there. 

A detail is that the ranges are start at 0, it should start at 1.
It would have to put much more coverage. 

Well but for this PR there is already enough,

**Comment by zbraniecki** [comment]
Thank you! This looks awesome!

**Pull request merged**